hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -954,7 +1009,31 @@ else if ( parentModelPart instanceof EntityCollectionPart ) {
 			}
 			return false;
 		}
-		if ( cardinality == Cardinality.MANY_TO_ONE ) {
+		else if ( parentNavigablePath.getParent() != null && creationState.resolveModelPart( parentNavigablePath.getParent() ) instanceof EmbeddedCollectionPart ) {//todo: handle recursively?
```

Comment:
We are still missing a test for this todo comment, but after that, we should be good to go. The test case would involve a `*ToOne` association mapping within an embeddable within an element collection. Also, the code comment in this if branch does not seem to fit the condition `instanceof EmbeddedCollectionPart` which makes it slightly confusing.